src/formatDate.js

```diff
@@ -9,6 +9,39 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+  const dataArr = date.split(fromSeparator);
```

Comment:
It's important to validate the `date` and `fromFormat` to ensure that they match in terms of the number of date parts and separators. If they don't match, the function should return an error or handle the discrepancy accordingly.